File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        int x=nums.size();
        while(x>0){
            int p=0;
            for(int i=0;i<nums.size();i++){
                if(nums[i]>=x){
                    p++;
                }
            }
            if(p==x){
                return x;
            }
            x--;
        }return -1;
        
    }
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.java
class Solution {
    public int specialArray(int[] nums) {
        int st=1,en=nums.length;
        while(st<=en){
            int x=st+(en-st)/2;
            int p=0;
            for(int i:nums){
                if(i>=x){
                    p++;
                }
            }
            if(p==x){
                return x;
            }
            else if(p>x){
                st=x+1;
            }
            else{
                en=x-1;
            }
            
        }return -1;
        
    }
}